subprojects/core/src/main/java/org/gradle/api/internal/file/DefaultFileOperations.java

```diff
@@ -200,4 +201,29 @@ public FileResolver getFileResolver() {
     public DefaultResourceHandler getResources() {
         return resourceHandler;
     }
+
+    public static DefaultFileOperations createSimple(FileResolver fileResolver, FileCollectionFactory fileTreeFactory, ServiceRegistry services) {
+        Instantiator instantiator = services.get(Instantiator.class);
+        FileLookup fileLookup = services.get(FileLookup.class);
+        FileSystem fileSystem = services.get(FileSystem.class);
+        Clock clock = services.get(Clock.class);
+        DirectoryFileTreeFactory directoryFileTreeFactory = services.get(DirectoryFileTreeFactory.class);
+        StreamHasher streamHasher = services.get(StreamHasher.class);
+        FileHasher fileHasher = services.get(FileHasher.class);
+        TextUrlResourceLoader.Factory textResourceLoaderFactory = services.get(TextUrlResourceLoader.Factory.class);
+        return new DefaultFileOperations(
+            fileResolver,
+            null,
+            null,
+            instantiator,
+            fileLookup,
+            directoryFileTreeFactory,
+            streamHasher,
+            fileHasher,
+            textResourceLoaderFactory,
+            fileTreeFactory,
+            fileSystem,
+            clock
+        );
+    }
```

Comment:
@eskatos The Kotlin DSL service logic now uses this method to create the `DefaultFileOperations` to basically attempt to unify on the creation of this type. Before there were multiple different ways that this object was created across three languages. Now, at least the Kotlin DSL and the `DefaultScript` for groovy are both using the same method.